### DISCUSSION TITLE
Proposal: Drop support for Cameras/Chimes/Intercoms

### DISCUSSION THREAD

**Discussion by tsightler** [tsightler/ring-mqtt#945] [General Discussion]
### Important Update

**Based on feedback, I have decided to keep this support for now.  We will see how it goes in 2025 and thanks to everyone for their feedback and support of this project.**

Hi All,

When this project first started it was called ring-alarm-mqtt and focused exclusively on Ring Alarm devices.  It became extremely popular with Home Assistant users where the native Ring integration did not support Ring Alarm (and I believe still does not).  While the native HA integration did support cameras/doorbells, it had significant issues, was not well maintained, had slow motion/ding updates to the use of polling instead of push notifications, and didn't support features such as lights and sirens on supported cameras, or other devices such as chimes and intercoms.  Because of this, I had lots of requests to support these devices and eventually I decided to include them and renamed the project to ring-mqtt.

After that I had even more requests to support live streaming from Ring cameras/doorbells and eventually I added an RTSP proxy service that allowed Ring cameras to be accessed via a standard protocol and stream live views in Home Assistant and other tools.

However, recently, the Ring integration in HA, as well as the underlying library, has a new maintainer that has been very active and implemented many great updates, including support for push notifications (same as ring-mqtt) and support for native live streaming directly in HA using the Ring WebRTC API (same as ring-mqtt uses to get the stream from Ring).  The integration also supports chimes and intercoms as well. 

Because of this, I'm seriously considering removing support for all "polled" Ring devices in the next major release of ring-mqtt.  This would include removing support for cameras, chimes, intercoms, and mode switching.  Support would remain for "hub" devices, which at this point is mostly just Ring Alarm, although the Smart Lighting bridge would remain supported for now.

Supporting polled devices, and especially live streaming, has been a significant development effort, but now that this is supported natively by the official Home Assistant Ring integration, which probably presents more than 95% of the users of this project, it seems there is little benefit to supporting these features in ring-mqtt and it would remove ~75% of the code and more troublesome upstream dependencies from the project.

I'm curious of the community reaction to this idea.

Thanks,
Tom

**Comment by iamenigmatic** [comment]
For what it is worth, I agree with this.  Building in redundancies doesn't make sense.  Things change, integrations evolve, and HA users should be used to this and ready for more tinker.  I would think your ultimate hope is an eventual full integration so you don't have to work on this anymore.

**Reply by tsightler** [reply]
Well, I wasn't even a user of HA when I started this project, it just kind of became the defacto tool for HA users with Ring Alarm when I was silly enough to add automatic HA MQTT discovery and put it on the HA forum and then someone else actually made the addon, but people kept coming here expecting me to help them figure out how to use it.  I expected maybe a few dozen "hacker" level users total, and now this project has nearly 10,000 users.

I'm not sure what I would do if HA natively supported Ring Alarm, maybe I would stop with it, but then again, I'm actually not a big HA user, I leverage MQTT and Node-Red to do pretty much everything as I find it far easier to work with, extremely flexible, and more consistently reliable.  HA just doesn't have the reliability I strive for in my home automation platform, and, quite honestly, I have a strong dislike of the automation rules in HA.  For super simple things, it's not bad, but any level of complexity/interdependency I find it way too complicated.

**Reply by tsightler** [reply]
I'm also super annoyed that HA 2024.11 basically broke generic camera support with ring-mqtt and the HA team just doesn't care at all.  I just don't find that they give one whit about usability.  Users have begged for various super simple to implement usability features for years and they just get into semantic arguments about why they don't want to do these things and thus they sit there undone or have to be implemented in various custom integrations and addons.  I've nearly given up trying to deal with them so my guess is I would keep the project but be relieved to tell HA users to just use the native integration.

**Reply by iamenigmatic** [reply]
I understand this completely.  I have always felt Red is very straight forward in what it does.  I have been using HA for over a decade and am used to the "extras" that need to be done to accomplish that reliability you are wanting.  It is excessive at times and requires coming at something in a way that doesn't seem intuitive.

**Reply by tsightler** [reply]
Note when I say "reliability" I mean due to breaking updates.  They've broken things with ring-mqtt three times just this year, and we're not talking rocket science here, just basic MQTT stuff.  But it's not even the breaking that is the problem, I get it, breakage happens, but it's the incredibly dismissive attitude when it does happen that really puts me off, I'm kind of tired of dealing with it.

**Reply by iamenigmatic** [reply]
> I'm also super annoyed that HA 2024.11 basically broke generic camera support with ring-mqtt and the HA team just doesn't care at all. I just don't find that they give one whit about usability. Users have begged for various super simple to implement usability features for years and they just get into semantic arguments about why they don't want to do these things and thus they sit there undone or have to be implemented in various custom integrations and addons. I've nearly given up trying to deal with them so my guess is I would keep the project but be relieved to tell HA users to just use the native integration.

I think they are extremely afraid to mess with Core.  My guess is they don't know what broke and how/why.  HA is the wild west IMO.  I have always tested new equipment and integrations because I am not afraid of breaking things and getting it running again.  I think they are like you in the sense that they have provided something initially but now have compounding support requests adding up.  On top of that, they now offer a paid service to login with, and that Core has become everything to them.  I think they are looking at it like every integration has to comply to their current ecosystem, and the integration must find a way to fix when the ecosystem changes.  Wild west.

**Reply by iamenigmatic** [reply]
> Note when I say "reliability" I mean due to breaking updates. They've broken things with ring-mqtt three times just this year, and we're not talking rocket science here, just basic MQTT stuff. But it's not even the breaking that is the problem, I get it, breakage happens, but it's the incredibly dismissive attitude when it does happen that really puts me off, I'm kind of tired of dealing with it.

Have you tested the latest updates to HA Core and OS to see if anything has been fixed?

**Reply by tsightler** [reply]
I wish that was the case, but they actually know exactly why it is broken, it would only require a couple of lines of code to fix, even I could produce the patch, and I can barely code in Python.  But they actually hide behind it being a design decision, regardless that it's terrible for usability and makes almost no sense.  There's been an outstanding request to change the underlying behavior for 6 years, or at least make it configurable, it would take a one line change to make it 95% more usable, and maybe 10 lines to make it 100% configurable if they are too afraid to change it globally for everyone (which is fair).
   
But that's what I mean about their approach, they care more about winning some semantic argument over why their original design decision was the correct one than listening to user feedback and changing their mind.  I was personally caught up in the ridiculous Iotawatt issue where they preferred to have a completely broken integration rather than do the right thing for users, even though it was patently clear what needed to happen.  To be clear, I'm sure they are good people doing their best, just don't enjoy dealing with the project as a whole.

**Reply by iamenigmatic** [reply]
Yeah, that is indeed garbage.  I have done projects where the support became so much that the project itself wasn't worth it any longer.  I go back to how much 10k of us appreciate the work you do.  I will use this as long as it is available to me.

**Reply by tsightler** [reply]
In the end, I feel like HA is kind of in their weird state.  When I did try to use it 5-6 years ago I liked it pretty much, but then they started trying to make it more "user friendly" however, everything they did to make it more user friendly, was mostly just annoying to me.  I used to be able to add 16 cameras in a config file, just copy the same lines over and over, making a minor tweak to each one, now it 16 trips through a wizard.  Want to move that same config to a dev system for testing, previously I could just copy the config, now, 16 more trips through the wizard or try to do some unsupported things.

However, while they try to make it easier for newbies, they really haven't fully succeeded at that and still ignore basic usability things.    So now HA is this weird state  that doesn't appeal to me as a geek that likes to tweak on things and do stuff with config files, but also doesn't really address the issues of newbies since they don't want to complicate things by making various things configurable, and then doesn't provide the ability to tweak things as needed for anybody.  

So then you end up with hacks like a zillion custom integrations that try to change the behavior in ways people have been asking for forever.  You can probably tell I'm really quite disillusioned with HA overall, but I admit it's probably the best solution for a whole host of people.

**Reply by KruseLuds** [reply]
> In the end, I feel like HA is kind of in their weird state. When I did try to use it 5-6 years ago I liked it pretty much, but then they started trying to make it more "user friendly" however, everything they did to make it more user friendly, was mostly just annoying to me. I used to be able to add 16 cameras in a config file, just copy the same lines over and over, making a minor tweak to each one, now it 16 trips through a wizard. Want to move that same config to a dev system for testing, previously I could just copy the config, now, 16 more trips through the wizard or try to do some unsupported things.
> 
> However, while they try to make it easier for newbies, they really haven't fully succeeded at that and still ignore basic usability things. So now HA is this weird state that doesn't appeal to me as a geek that likes to tweak on things and do stuff with config files, but also doesn't really address the issues of newbies since they don't want to complicate things by making various things configurable, and then doesn't provide the ability to tweak things as needed for anybody.
> 
> So then you end up with hacks like a zillion custom integrations that try to change the behavior in ways people have been asking for forever. You can probably tell I'm really quite disillusioned with HA overall, but I admit it's probably the best solution for a whole host of people.

Do what you gotta do :-(

**Comment by KruseLuds** [comment]
@tsightler thank you for all you do for this for us!

**Comment by KruseLuds** [comment]
How does removing this functionality affect a scenario where someone relies upon it only to notify HA of movement caught by the Ring Doorbell v2?

**Reply by tsightler** [reply]
Ring and Nest are completely different companies so, not at all I guess?  If you mean Ring Doorbell V2, then you would just use the native Home Assistant Ring integration.

**Reply by KruseLuds** [reply]
Yes I changed it to Ring, my bad. I actually use motion detection from both Ring-MQTT and the ring integration as triggers for the same automations, as Ring-MQTT would sense the motion much more quickly than the Ring integration does. However it's just the Ring integration sensing motion now, still very slowly as usual... :-(

**Reply by tsightler** [reply]
I don't know why the current HA native Ring integration would be slower than ring-mqtt as it now uses the identical method for detecting events.  I've tested them head-to-head and the difference is effectively not measurable, so it's a bit strange that it is slow for you.   It makes me think that something in your network is blocking connections to the push server because that would keep ring-mqtt from being able to receive notifications and would force the native integration to revert to the old school polling method.

**Comment by nebhead** [comment]
I say go with your gut.  I'll use both integrations and be completely happy.  The camera's were secondary for me anyway, as I was only really relying on the alarm piece of the integration - which works beautifully I might add.

**Reply by tsightler** [reply]
I'm still considering some other options.  Some of the frustration is that I keep hitting bugs in the upstream products that I can't get fixed, even when I confirm the issues and submit PRs for them myself.  I have to ship a custom version of go2rtc, and now HA 2024.11 and later have made changes that really negatively impacts the use with ring-mqtt, but they don't seem to care, even though they kind of acknowledge the problem, and even admit it should be improved.  Push receiver has multiple bugs, the maintainer there is at least a bit responsive, but bugs still remain unfixed for weeks/months.

**Comment by KruseLuds** [comment]
Interesting, the plot thickens! Thanks for the note

**Comment by jamesonuk** [comment]
I use this for Alarm only.   I did use it for cameras at one point but even with your great work it seems it is always fighting against Ring who don't want you to use the cameras outside of their ecosystem. 

I have replaced my Ring cameras and not looked back.   I suspect their recent huge price rises have caused many HA users to do the same.

Most references I see to this on forums reference alarms and that bit works great.   If removing all the fluff needed to deal with getting cameras to work and energy for polling makes your life easier and the code cleaner I would say do it.    Users who are using it now can still stay on the current version until they migrate,

**Comment by jeffgraham-hub** [comment]
I for one will miss the support for Cameras. I only use live view occasionally. I do use ring-mqtt to push the received motion snapshots to all the TVs in the house. So much easier than having to pick up the phone to see who is at the front door. However, I do understand your frustration and what must be a fair amount of your time spent supporting this feature. I haven't look at the native RIng support in HA since finding this project but I guess I may need to.

**Reply by tsightler** [reply]
Nothing is decided for sure yet, it's a proposal, nothing more at this point.  Also, next major version is not even planned right now.  Maybe things will stabilize in the coming months and make this post obsolete.

**Comment by pietje63** [comment]
Thanks for all the good work!

Just my personal position. I use this addon to store ring videos of a ring camera without subscription. That won't work with the official integration, will it? Will really miss that feature if it is removed.

Fully understand that maintaining it is a lot of work! If there are more like me, it may make sense to split the add on and hopefully we find another dev for the same use I have.

**Reply by tsightler** [reply]
Thanks for the feedback.  While I understand this use case, it's actually one of the reasons I was considering dropping camera support.  As Ring raises prices, more and more users seem to be seeking the project for this use case.  I don't really have an issue with users doing whatever they want, but, in the real world, if a significant number of users leverage this project for that use case, it increases the changes that Ring would just block projects like this completely, similar to MyQ.  

That being said, I don't see why the native integration would not allow recording videos since it now has live streaming support, but I'll admit to not trying it.

**Reply by nebhead** [reply]
TBH - I just took advantage of holiday sales to pick up the Reolink doorbell precisely because Ring is raising subscription pricing.  While I'm grandfathered in for the next year for the original pricing, I think it's the end of the road for me with regards to the Ring cameras. (I am still an alarm user)  

So far, I'm really liking the new doorbell which is incredibly snappy and well supported by Home Assistant - even allowing me to do similar notifications, all locally, no subscription required.  If companies aren't willing to support open APIs (even in the cloud) for a small audience of power-users, then my money will likely be taken elsewhere.

**Comment by leccelecce** [comment]
Thank you for all your endeavours. I for one would rather have an application that does one thing well, and can be sustainably maintained, than tries to do a lot and risks falling away.

**Comment by Blubbaz** [comment]
@tsightler Big thanks for your amazing work! As a developer myself, I understand and support your interest in preventing redundant work and maintaining effort. However, I would kindly ask that, if you decide to drop support, you continue supporting the intercom until the official HASS add-on fixes the Ring intercom integration to be properly recognized as a lock, as your add-on currently does. Without that proper recognition, it is not possible to integrate and control the intercom in HomeKit.

https://github.com/home-assistant/core/issues/132061

**Comment by ChrisIsBananaz** [comment]
I was poking around looking for clues as to why I could not see some ding and any unlock messages from my Intercom in the integration. 

I did not know there was an official Ring integration separate from this one. There's so much ground truth out on all forums and communities pointing to this integration. So, I'd ask that there's some multi-community warnings at least 2 releases before the action is taken so some of us can migrate non-alarm devices to the other integration. I rarely go looking in Github for integration updates and release notes just before an update doesn't give much notice. Please ignore me if that was something you were planning on already.

On another note, I saw you were looking for Intecom users to test something. If you haven't found folks, I'd be happy to help. I'm one of the -some unknown but probably small number- people to have one working in the US.

**Comment by mat-pandaz** [comment]
The Ring offical app right now doesn't handle Intercom well.  I currently use this addon to get the call request (when someone pushes my intercom button) and without it I wouldn't have integration.  The official one uses events and I'm not sure how to deal with that whereas yours uses a binary sensor which was very easy to set up.

**Reply by tsightler** [reply]
Assuming you mean the Home Assistant native Ring integration, using events is probably the correct way, binary sensor is a bit of a hack because there wasn't a way to create an event in HA via MQTT (as far as I know it's still not possible).  Events are quite easy to use as triggers, requires just the event name and that's it.

That being said, for now I've decided to keep support around.  We'll see where it goes in the future, after all the recent changes maybe there will be a good period of stability for the Ring API.  In the end, I consider the changes they've made improvements, and I think the signaling model they have implemented and the new push notification format a significant enhancement, so maybe the changes will die down a bit.

And, to be clear, I'm not complaining about Ring making changes, it's their service after all, but that doesn't change the reality that keeping up with those changes require time I mostly don't have.  I spent untold personal hours reverse engineering all the changes, pushing them into upstream projects, and tracking down various bugs in some of those upstream projects as well.  Those are just hours I don't have in abundance at this point in my life.  I enjoy doing it, and if I just had nothing but time, it would be a different scenario.

**Reply by kassandra79** [reply]
@tsightler thank you so much for this integration. I am completely new to Home Assistant and don't really know how to code without help. I have a Ring Doorbell, an intercom and a StickUp Camera and it only took me a few hours as a newbie to configure everything so I get a notification to my phone with a snapshot from the live stream and option to either open the door or go to live view in HA. I know this would be easy to do with Ring if I used their subscription, but subscriptions keep piling up so I am always looking for workarounds. And your great add-on really does the trick. 

I came up with this code

`alias: Building door notification
description: ""
triggers:
  - trigger: state
    entity_id:
      - binary_sensor.building_door_ding
    from: "off"
    to: "on"
conditions: []
actions:
  - type: turn_on
    device_id: Xxx
    entity_id: Xxx
    domain: switch
  - delay:
      hours: 0
      minutes: 0
      seconds: 2
      milliseconds: 0
  - action: camera.snapshot
    metadata: {}
    data:
      filename: /config/www/ring_last.jpg
    target:
      entity_id: camera.ring_live_snapshot
  - delay:
      hours: 0
      minutes: 0
      seconds: 1
      milliseconds: 0
  - action: notify.mobile_app_name
    metadata: {}
    data:
      message: Zum Livestream tippen
      title: Es klingelt an der Eingangstür
      data:
        image: >-
          https://mynabucasaID.ui.nabu.casa/local/ring_last.jpg?{{
          now().timestamp() | int }}
        clickAction: /doors-cameras/building-door
        actions:
          - action: URI
            title: Livestream öffnen
            uri: /doors-cameras/building-door
          - action: building_door_unlock
            title: Tür öffnen
  - delay:
      hours: 0
      minutes: 0
      seconds: 30
      milliseconds: 0
  - type: turn_off
    device_id: Xxx
    entity_id: Xxx
    domain: switch
mode: single`

I do have one small problem with triggering the automation - but using the official Ring integration does not solve my problem because the event it records in the logs cannot trigger an automation apparently. 

My automation uses the binary ding sensor and works just fine when I simulate it in dev tools but when the actual door bell (intercom) is pressed, I do see the event in the Ring integration but not the ding sensor changing states to on in the MQTT log as it does in my tests. If I understood correctly, I can’t use the event from Ring to trigger this automation.

That same code works for the doorbell right now, but the doorbell's states are occupied and clear. I am not 100% sure if it just works because I had to activate a trial subscription and if it will still continue to work once that subscription is cancelled. 

@mat-pandaz @jeffgraham-hub I read that you both use the intercom to trigger automations. Could you let me know what you use as a trigger?
